docs/tutorial/writing-tests/test-case.rst

```diff
@@ -0,0 +1,1105 @@
+As you can see, checkbox presents the following result:
+
+.. code-block:: none
+
+    [...]
+    ==================================[ Results ]===================================
+     ☑ : Test that the internet is reachable
+     ☑ : Test that the network speed is acceptable
+
+If asked to run a job that depends on another job, Checkbox will try to pull
+the other job and its dependencies automatically. If Checkbox is unable to do
+so we can always force this behavior by listing the jobs in order of dependence
+in the run command:
+
+.. code-block:: none
+
+    (checkbox_venv) > checkbox-cli run com.canonical.certification::network_available \
+      com.canonical.certification::network_speed
+
+Finally lets test that this actually works. To do so we can temporarily change the
+command section of ``network_available`` to ``exit 1``. This
+is the new Result that Checkbox will present:
+
+.. code-block:: none
+
+    [...]
+    -----------------[ Test that the network speed is acceptable ]------------------
+    ID: com.canonical.certification::network_speed
+    Category: com.canonical.plainbox::uncategorised
+    Job cannot be started because:
+      - required dependency 'com.canonical.certification::network_available' has failed
+    Outcome: job cannot be started
+    Finalizing session that hasn't been submitted anywhere: checkbox-run-2024-08-02T13.31.58
+    ==================================[ Results ]===================================
+     ☒ : Test that the internet is reachable
+     ☐ : Test that the network speed is acceptable
+
+Customize tests via envvars
+===========================
+
+Sometimes it is hard to set an unique value for a test parameter because it may
+depend on a multitude of factors. Notice that our previous test has a very
+ISP-generous interpretation of what is an acceptable speed, some customers may
+beg to differ. At the same time it is hard to define an acceptable speed for
+any interface and all machines. In Checkbox we use environment variables
+to customize testing parameters that have to be defined per-machine/test run.
+Consider the following:
+
+.. code-block:: none
+
+    id: network_speed
+    flags: simple
+    _summary: Test that the network speed is acceptable
+    environ:
+      ACCEPTABLE_BYTES_PER_SECOND_SPEED
+    command:
+      echo Testing for the limit speed: ${ACCEPTABLE_BYTES_PER_SECOND_SPEED:-600}
+      curl -Y ${ACCEPTABLE_BYTES_PER_SECOND_SPEED:-600} -o /dev/null \
+        https://cdimage.ubuntu.com/ubuntu-mini-iso/noble/daily-live/current/noble-mini-iso-amd64.iso
+
+Before running the test we have to define a Checkbox configuration. Note that
+if we were using a test plan, we could run it with a launcher, but the
+``run`` command doesn't take a launcher parameter, so we have to use a
+configuration file. Place the following in ``~/.config/checkbox.conf``.
+
+.. code-block:: ini
+
+    [environment]
+    ACCEPTABLE_BYTES_PER_SECOND_SPEED=60000000
```

Comment:
I see `60000000` here, `600` as the default in the job, and I'm a bit confused, especially as `-Y` is defined as

```
       -Y, --speed-limit <speed>
              If a transfer is slower than this given speed (in bytes per sec‐
              ond) for speed-time seconds it gets aborted. speed-time  is  set
              with -y, --speed-time and is 30 if not set.
```

Reply:
(ops, -y is 30 by default, I thought it was 1)